design/one-pager-generic-datasource.md

```diff
@@ -0,0 +1,171 @@
+# Patch from Generic Datasources
+
+* Owner: Maximilian Blatt (maximilian.blatt@accenture.com, @mistermx)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## 1 Background
+
+Crossplane currently supports patching from composite resources to composed
+resources and vice versa. It is also possible to copy values between two
+composed resources inside one compositiion by patching the value from one
+composed back to the composite and then to the second composed resource.
+
+However, Crossplane currently does not provide a way to patch values from
```

Comment:
Could you expand this with more specifics, i.e. what is _dynamic_ resource? An example use case would also be great.

Reply:
I refer to everything that is hardcoded in the composition as *static*. This includes patches between the composite and the composed resources. Its *static* because they behave the same regardless of the environment.

By *dynamic* I mean environment-dependent values. I.e. deploying AWS resources in two different VPCs requires different subnet IDs, VPC Ids, OIDC issuers etc.

Reply:
Makes sense. I dropped `dynamic` and `static` and use `environment-dependent patching` instead.